libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -2622,6 +2622,61 @@ - (BOOL)isPossibleNumber:(NBPhoneNumber*)number
 }
 
 
+- (NBEValidationResult)testNumberLength:(NSString *)number metadata:(NBPhoneMetaData *)metadata {
+    return [self testNumberLength:number metadata:metadata type:NBEPhoneNumberTypeUNKNOWN];
+}
+
+- (NBEValidationResult)testNumberLength:(NSString *)number metadata:(NBPhoneMetaData *)metadata type:(NBEPhoneNumberType)type {
+    NBPhoneNumberDesc *descForType = [self getNumberDescByType:metadata type:type];
+    
+    NSArray<NSNumber *> *possibleLengths = [descForType.possibleLength count] == 0 ? metadata.generalDesc.possibleLength : descForType.possibleLength;
+    
+    NSArray<NSNumber *> *localLengths = descForType.possibleLengthLocalOnly;
+    
+    if (type == NBEPhoneNumberTypeFIXED_LINE_OR_MOBILE) {
+        if ([self descHasPossibleNumberData:[self getNumberDescByType:metadata type:NBEPhoneNumberTypeFIXED_LINE]]) {
+            return [self testNumberLength:number metadata:metadata type:NBEPhoneNumberTypeMOBILE];
+        } else {
+            NBPhoneNumberDesc *mobileDesc = [self getNumberDescByType:metadata type:NBEPhoneNumberTypeMOBILE];
+            if([self descHasPossibleNumberData:mobileDesc]) {
+                possibleLengths = [[possibleLengths arrayByAddingObjectsFromArray:[mobileDesc.possibleLength count] == 0 ? metadata.generalDesc.possibleLength : mobileDesc.possibleLength] sortedArrayUsingComparator:^NSComparisonResult(id  _Nonnull obj1, id  _Nonnull obj2) {
+                    return obj1 < obj2;
+                }];
+                
+                if (![localLengths count]) {
+                    localLengths = mobileDesc.possibleLengthLocalOnly;
+                } else {
+                    localLengths = [[localLengths arrayByAddingObjectsFromArray:mobileDesc.possibleLengthLocalOnly] sortedArrayUsingComparator:^NSComparisonResult(id  _Nonnull obj1, id  _Nonnull obj2) {
+                        return obj1 < obj2;
+                    }];
+                    
+                }
+            }
+        }
+    }
+    
+    if ([possibleLengths.firstObject isEqualToNumber:@-1]) {
```

Comment:
can you put parens around the -1? Also here you use isEqualToNumber, and below you use .integerValue ==. Can we be consistent on how we do number equality?